pkg/cmdimpl/add.go

```diff
@@ -349,134 +270,276 @@ func Add(args []string, allParams AddParamSet, deps *AddDependencies) error {
 	case status.Unmodified:
 		return errors.New("WeGO not installed... exiting")
 	case status.Unknown:
-		return errors.New("WeGO can not define cluster status... exiting")
+		return errors.New("WeGO can not determine cluster status... exiting")
 	}
 
-	// Set up wego repository if required
-	wegoRepoName, err := fluxops.GetRepoName()
+	switch strings.ToUpper(params.AppConfigUrl) {
+	case string(ConfigTypeNone):
+		return addAppWithNoConfigRepo()
+	case string(ConfigTypeUserRepo):
+		return addAppWithConfigInUserRepo(ctx, deps.GitClient)
+	default:
+		return addAppWithConfigInExternalRepo(ctx, deps.GitClient)
+	}
+}
+
+func addAppWithNoConfigRepo() error {
+	// Source covers entire user repo
+	userSourceName := getUserRepoName()
+	userRepoSource, err := generateSource(userSourceName, params.Url, SourceTypeGit)
+	if err != nil {
+		return wrapError(err, "could not set up GitOps for user repository")
+	}
+	appYaml, err := generateAppYaml()
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create app.yaml for '%s'", params.Name))
+	}
+	// kustomize or helm referencing single user repo source
+	applicationGOAT, err := generateApplicationGoat(userSourceName)
 	if err != nil {
-		return wrapError(err, "could not get flux repo name")
+		return wrapError(err, fmt.Sprintf("could not create GitOps automation for '%s'", params.Name))
```

Comment:
error message is using a different value than what is passed to generateApplicationGoat